uk.co.compucorp.civicrm.tasksassignments/api/v3/TASettings.php

```diff
@@ -1,70 +1,161 @@
 <?php
 
+/**
+ * TASettings.get API specification
+ *
+ * @param array $spec
+ */
+function _civicrm_api3_t_a_settings_get_spec(&$spec) {
+  $spec['fields'] = [
+    'name' => 'fields',
+    'title' => 'Fields',
+    'description' => 'Which field(s) you want returned',
+    'type' => CRM_Utils_Type::T_STRING,
+    'api.required' => 0,
+  ];
+}
+
+/**
+ * TASetting.get API
+ *
+ * Used to get one or all of the T&A settings.
+ *
+ * When used without any params, all the settings will be
+ * returned. Otherwise, if the `fields` param is present,
+ * only the given fields will be returned.
+ *
+ * @param array $params
+ *
+ * @return array
+ * @throws \CiviCRM_API3_Exception
+ */
 function civicrm_api3_t_a_settings_get($params) {
-    
-    $fields = array(
-        'documents_tab',
-        'keydates_tab',
-        'add_assignment_button_title',
-        'number_of_days',
-        'auto_tasks_assigned_to',
-        'is_task_dashboard_default',
-        'days_to_create_a_document_clone',
-    );
-    
-    if (!empty($params['fields'])) {
-        $fields = (array)$params['fields'];
-    }
-    
-    $result = array();
-    
-    foreach ($fields as $field) {
-        $setting = civicrm_api3('OptionValue', 'get', array(
-            'option_group_id' => 'ta_settings',
-            'name' => $field,
-        ));
-        $item =  CRM_Utils_Array::first($setting['values']);
-        $result[$field] = array(
-            'name' => $item['name'],
-            //'label' => $item['label'],
-            'value' => $item['value'] != "null" ? $item['value'] : ""
-        );
+  $fields = _civicrm_api3_t_a_settings_valid_fields();
+
+  if (!empty($params['fields'])) {
+    $fields = (array) $params['fields'];
+  }
+
+  $settings = _civicrm_api3_t_a_settings_get_settings();
```

Comment:
You only use ` $settings['ta_settings']` after this, so how about shortening the future variables and making `$settings = _civicrm_api3_t_a_settings_get_settings()['ta_settings'];`?